Sources/XCTest/XCTestCase.swift

```diff
@@ -98,4 +129,137 @@ extension XCTestCase {
+    ///   expectations are not met before the given timeout. Default is the file
+    ///   containing the call to this method. It is rare to provide this
+    ///   parameter when calling this method.
+    /// - Parameter line: The line number to use in the error message if the
+    ///   expectations are not met before the given timeout. Default is the line
+    ///   number of the call to this method in the calling file. It is rare to
+    ///   provide this parameter when calling this method.
+    /// - Parameter handler: If provided, the handler will be invoked both on
+    ///   timeout or fulfillment of all expectations. Timeout is always treated
+    ///   as a test failure.
+    ///
+    /// - Note: Whereas Objective-C XCTest determines the file and line
+    ///   number of the "wait" call using symbolication, this implementation
+    ///   opts to take `file` and `line` as parameters instead. As a result,
+    ///   the interface to these methods are not exactly identical between
+    ///   these environments.
+    public func waitForExpectationsWithTimeout(timeout: Double, file: StaticString = __FILE__, line: UInt = __LINE__, handler: XCWaitCompletionHandler?) {
+        // Mirror Objective-C XCTest behavior; display a test failure when
+        // users wait without having first set expectations.
+        if XCTAllExpectations.count == 0 {
+            let failure = XCTFailure(
+                message: "call made to wait without any expectations having been set",
+                failureDescription: "API violation",
+                expected: false,
+                file: file,
+                line: line)
+            XCTAllExpectationFailures.append(failure)
+            return
+        }
+
+        // Objective-C XCTest outputs the descriptions of every unfulfilled
+        // expectation. We gather them into this array, which is also used
+        // to determine failure--a non-empty array meets expectations weren't
+        // met.
+        var unfulfilledDescriptions = [String]()
+
+        // We continue checking whether expectations have been fulfilled until
+        // the specified timeout has been reached.
+        let runLoop = NSRunLoop.currentRunLoop()
+        let timeoutDate = NSDate(timeIntervalSinceNow: timeout)
+        while NSDate().compare(timeoutDate) == NSComparisonResult.OrderedAscending {
+            unfulfilledDescriptions = []
+            for expectation in XCTAllExpectations {
+                if !expectation.fulfilled {
+                    unfulfilledDescriptions.append(expectation.description)
+                }
+            }
+
+            // If we've met all expectations, then break out of the specified
+            // timeout loop early.
+            if unfulfilledDescriptions.count == 0 {
+                break
+            }
+
+            // Otherwise, wait another fraction of a second.
+            runLoop.runUntilDate(NSDate(timeIntervalSinceNow: 0.01))
```

Comment:
Looks like the above works perfectly fine on Linux. Perhaps swift-corelibs-xctest should depend upon Objective-C Foundation when being built for OS X, instead of SwiftFoundation?